### PR TITLE
Update solution to use multi-target

### PR DESCRIPTION
**Pull request by ravinderjangra**
**Changes:**
- Move test projects to the Tests directory
- Convert `SafeApp.AppBindings` and `SafeApp.MockAuthBindings` to use multi-target
- Update `.nuspec` file
- Update cake script to support new project structure

**Note:**
- CI build may fail as it'll require some changes

### CODE REVIEW

**Comment by ravinderjangra** [comment]
If CI fails then I will create a new CI pipeline for this branch

**Comment by ravinderjangra** [comment]
closing and reopening to run AzureDevOps CI

File: SafeApp.AppBindings/SafeApp.AppBindings.csproj [hunk 1]
```diff
@@ -1,13 +1,60 @@
...
+
+  <ItemGroup>
+    <AndroidNativeLibs Include="NativeLibs\Android\**\*.so" />
+    <iOSSafeAppLibs Include="NativeLibs\iOS\**\*.a" />
+    <DesktopNativeLibs Include="NativeLibs\Desktop\**\*.*" />
+  </ItemGroup>
+
+  <Target Name="CheckNativeLibs" BeforeTargets="DispatchToInnerBuilds">
+    <PropertyGroup>
+      <AndroidLibsExists Condition=" '@(AndroidNativeLibs-&gt;Count())' == '0' ">true</AndroidLibsExists>
```
Comment (oetyng) [inline SafeApp.AppBindings/SafeApp.AppBindings.csproj]: It works as is, but I think this is slightly confusing:
Shouldn't [*]LibsExists be false when the count is zero?
And by that, further down, DownloadNativeLibs is true when any of them are false.

File: CakeHelperScripts/DesktopTest.cake [hunk 2]
```diff
@@ -20,17 +20,18 @@ Task("Build-Desktop-Project")
...
     var osFamily = (int)Context.Environment.Platform.Family;
     if(osFamily == 1)
-      dotnetBuildArgument = @"--runtime win10-x64";
+      dotnetBuildArgument = @"-p:RuntimeIdentifiers=win10-x64";
     else if(osFamily == 2)
-      dotnetBuildArgument = @"--runtime linux-x64";
+      dotnetBuildArgument = @"-p:RuntimeIdentifiers=linux-x64";
     else if (osFamily == 3)
-      dotnetBuildArgument = @"--runtime win-x64";
+      dotnetBuildArgument = @"-p:RuntimeIdentifiers=win-x64";
```
Comment (oetyng) [inline CakeHelperScripts/DesktopTest.cake]: Hm, is this right? osFamily == 3 would correspond to OSX according to https://github.com/cake-build/cake/blob/develop/src/Cake.Core/PlatformFamily.cs

Comment (ravinderjangra) [reply CakeHelperScripts/DesktopTest.cake]: Ohh, I will update this

File: SafeApp.AppBindings/SafeApp.AppBindings.csproj (hunk 1 above)
Comment (ravinderjangra) [reply SafeApp.AppBindings/SafeApp.AppBindings.csproj]: That makes sense, I will add that in issues/todo list and then we can pick it up in a separate PR.

**Comment by ravinderjangra** [comment]
vNext devOps CI: https://dev.azure.com/maidsafe/SafeApp/_build/results?buildId=848&view=results

**Pull request merged**